hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java

```diff
@@ -996,7 +996,7 @@ public void notifyNotLeader(Collection<TransactionContext> pendingEntries) {
 
   @Override
   public CompletableFuture<Void> truncate(long index) {
-    stateMachineDataCache.removeIf(k -> k >= index);
+    stateMachineDataCache.removeIf(k -> k <= index);
```

Comment:
@sumitagrawl , For `truncate`, it should be `k > index`, i.e. removed the truncated entries.

The change looks good other than that.

Reply:
@szetszwo Thanks for the review.  When requesting changes I think it's better to not start CI checks, since the update will require a new run anyway.  For the current state of the patch we can see results in @sumitagrawl's [fork](https://github.com/sumitagrawl/ozone/actions/runs/4561460466).

Reply:
@szetszwo Updated...